lib/commands/actions.js

```diff
@@ -166,6 +167,16 @@ commands.getScreenshot = async function () {
     await fs.unlink(localFile);
   }
   await this.adb.pull(png, localFile);
+  let screenOrientation = await this.adb.getScreenOrientation();
+  if (screenOrientation) {
+    jimp.read(localFile).then(function (image) {
```

Comment:
I have rewritten it to use `jimp.getBuffer` instead of writing to disk and re-reading. My problem is that `jimp.write` and `jimp.getBuffer` don't return promises, but instead receive a callback. What do you recommend to do a call to `jimp.getBuffer` and wait until its callback has been triggered? Is there a suggested way to do this in Appium?

Currently, my code reads:
```js
  let screenOrientation = await this.adb.getScreenOrientation();
  let image = await jimp.read(localFile);
  try {
    image = await image.rotate(90 * screenOrientation);
  } catch (err) {
    log.warn(`Could not rotate screenshot due to error: ${err}`);
  }
  let b64data = // Some way to wait for image.getBuffer(jimp.MIME_PNG, function () {}) to finish and call toString('base64') on the data
```

Thanks for the help, JS isn't a language I am used to.